core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -694,7 +694,25 @@ public void onClosed(Channel c, IOException cause) {
         old = ACL.impersonate(ACL.SYSTEM);
         try {
             for (ComputerListener cl : ComputerListener.all()) {
-                cl.onOnline(this,taskListener);
+                try {
+                    cl.onOnline(this,taskListener);
+                } catch (IOException | InterruptedException e) {
+                    // Per Javadoc log exceptions but still go online.
+                    taskListener.getLogger().format(
+                        "onOnline: %s reported an exception: %s%n",
+                        cl.getClass(),
+                        e.toString());
+                } catch (Exception e) {
```

Comment:
I can't detect a difference between this catch and the prior one starting on line 699. Is there any reason not to combine them together?

Reply:
I kept them separate because the API specifically names these two types.  I can unify them sure.